google/ads/google_ads/client.py

```diff
@@ -267,16 +270,18 @@ def _get_request_id(self, trailing_metadata):
         return None
 
     def _handle_grpc_exception(self, exception):
-        """Handles gRPC exceptions of type RpcError by attempting to
-           convert them to a more readable GoogleAdsException. Certain types of
-           exceptions are not converted; if the object's trailing metadata does
-           not indicate that it is a GoogleAdsException, or if it falls under
-           a certain category of status code, (INTERNAL or RESOURCE_EXHAUSTED).
-           See documentation for more information about gRPC status codes:
-           https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+        """Attempts to convert exceptions to a GoogleAdsException object.
+
+        Handles gRPC exceptions of type RpcError by attempting to convert them
+        to a more readable GoogleAdsException. Certain types of exceptions are
+        not converted; if the object's trailing metadata does not indicate that
+        it is a GoogleAdsException, or if it falls under a certain category of
+        status code, (INTERNAL or RESOURCE_EXHAUSTED). See documentation for
+        more information about gRPC status codes:
+        https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
 
         Args:
-            exception: an exception of type RpcError.
+            exception: an exception of type grpc.RpcError.
```

Comment:
Just so I understand, is the best way to check whether a request has failed to inspect that `response.exception()` returns a value? And then if there's a value, we should treat it as a `grpc.Call` instance? Or is there another mechanism that's better?

Reply:
And is it fair to refer to the values returned by `continuation` as `grpc.Call`/`grpc.Future` since the objects inherit from both classes? Or is it better to just use `grpc.Call`?

Reply:
If there is a value returned by `response.exception()`, then the call failed and `response` should be used to inspect error details. `response` is both a `Call` and a `Future`, so it's appropriate to refer to it as both. 

The call to `exception()` returns an `RpcError` in the event of an rpc failure, which will usually (when generated by gRPC library) actually just be `response` itself, e.g., also a `Call` and a `Future` (we should probably update our documentation to guarantee this, but it is currently not guaranteed). However, if the exception occurred elsewhere, e.g., thrown by the interceptor logic provided by the application, then `response.exception()` will just return the thrown exception, which can be of any type and will (for once) probably not be a `Call` or `Future`.

So, the return value of `response.exception()` should only be used if you're checking that it is some other type besides/beyond `RpcError`, such as a custom exception thrown by your application layer. For any gRPC information on the failure, the methods on `response` should be used directly.

Reply:
Resolve this for this PR but will revisit in a separate PR.